File: create_admin.js
require("dotenv").config();
const mongoose = require("mongoose");
const bcrypt = require("bcryptjs");

const MONGO_URI = process.env.MONGODB_URI;

const UserSchema = new mongoose.Schema({}, { strict: false });
const User = mongoose.model("User", UserSchema, "users");

async function createAdmin() {
  try {
    await mongoose.connect(MONGO_URI);
    console.log("✅ Connected to MongoDB");

    const existing = await User.findOne({ role: "admin" });

    if (existing) {
      console.log("⚠️ Admin already exists.");
      process.exit(0);
    }

    const hashedPassword = await bcrypt.hash("PalawanPhillip90!", 10);

    await User.create({
      username: "palawan_admin",
      email: "admin@palawandelivery.ph",
      password: hashedPassword,
      role: "admin",
      createdAt: new Date()
    });

    console.log("🎉 SUCCESS!");
    console.log("Username: palawan_admin");
    console.log("Password: PalawanPhillip90!");

    process.exit(0);

  } catch (err) {
    console.error("❌ ERROR:", err.message);
    process.exit(1);
  }
}

createAdmin();

File: create_admin.py
from pymongo import MongoClient
from werkzeug.security import generate_password_hash

# ✅ FINAL WORKING LINK — check password is correct
MONGODB_URI = "mongodb+srv://phillipbadajos44_db_user:Fskyae72oaNbeFAv@cluster0.mt8uncj.mongodb.net/delivery_db?retryWrites=true&w=majority&appName=Cluster0"
client = MongoClient(MONGODB_URI, serverSelectionTimeoutMS=20000)
db = client.delivery_db

admin = {
    "username": "palawan_admin",
    "email": "admin@palawandelivery.ph",
    "password": generate_password_hash("PalawanPhillip90!", method="pbkdf2:sha256"),
    "role": "admin"
}

# Check if admin already exists
existing = db.users.find_one({"role": "admin"})
if existing:
    print("⚠️ Admin already exists — no new one added.")
else:
    db.users.insert_one(admin)
    print("✅ SUCCESS! Admin account created.")
    print("🔐 Username: palawan_admin")
    print("🔐 Password: PalawanPhillip90!")

client.close()
